tests/Unit/Evolution/DiscontinuousGalerkin/SlopeLimiters/Test_Krivodonova.cpp

```diff
@@ -0,0 +1,2900 @@
+  using Limiter = Krivodonova<dim, tmpl::list<ScalarTag<0>>>;
+  // Use non-unity (because that's the default) but close alpha values to make
+  // the math easier but still test thoroughly.
+  Limiter krivodonova{std::vector<double>(
+      Spectral::maximum_number_of_points<Spectral::Basis::Legendre>, 0.99)};
+
+  NeighborData<dim, typename Limiter::PackagedData> neighbor_data{};
+
+  const Element<dim> element(ElementId<dim>{0}, {});
+  // We don't care about the ElementId for these tests, just the direction.
+  Limiter::PackagedData& package_data_up_xi = neighbor_data[std::make_pair(
+      Direction<dim>::upper_xi(), ElementId<dim>{0})];
+  Limiter::PackagedData& package_data_lo_xi = neighbor_data[std::make_pair(
+      Direction<dim>::lower_xi(), ElementId<dim>{0})];
+  Limiter::PackagedData& package_data_up_eta = neighbor_data[std::make_pair(
+      Direction<dim>::upper_eta(), ElementId<dim>{0})];
+  Limiter::PackagedData& package_data_lo_eta = neighbor_data[std::make_pair(
+      Direction<dim>::lower_eta(), ElementId<dim>{0})];
+
+  package_data_up_xi.modal_volume_data.initialize(num_pts);
+  package_data_up_xi.mesh = mesh;
+  package_data_lo_xi.modal_volume_data.initialize(num_pts);
+  package_data_lo_xi.mesh = mesh;
+  package_data_up_eta.modal_volume_data.initialize(num_pts);
+  package_data_up_eta.mesh = mesh;
+  package_data_lo_eta.modal_volume_data.initialize(num_pts);
+  package_data_lo_eta.mesh = mesh;
+
+  Scalar<DataVector> nodal_scalar_data_to_limit(num_pts, 0.0);
+  DataVector expected(num_pts);
+  const auto helper =
+      [
+        &element, &expected, &krivodonova, &mesh, &neighbor_data,
+        &package_data_lo_xi, &package_data_up_xi, &package_data_lo_eta,
+        &package_data_up_eta, &nodal_scalar_data_to_limit
+      ](const ModalVector& up_xi_coeffs, const ModalVector& up_eta_coeffs,
+        const ModalVector& initial_coeffs, const ModalVector& lo_xi_coeffs,
+        const ModalVector& lo_eta_coeffs,
+        const ModalVector& expected_coeffs) noexcept {
+    to_nodal_coefficients(&get(nodal_scalar_data_to_limit), initial_coeffs,
+                          mesh);
+    get(get<::Tags::Modal<ScalarTag<0>>>(
+        package_data_up_xi.modal_volume_data)) = up_xi_coeffs;
+    get(get<::Tags::Modal<ScalarTag<0>>>(
+        package_data_lo_xi.modal_volume_data)) = lo_xi_coeffs;
+    get(get<::Tags::Modal<ScalarTag<0>>>(
+        package_data_up_eta.modal_volume_data)) = up_eta_coeffs;
+    get(get<::Tags::Modal<ScalarTag<0>>>(
+        package_data_lo_eta.modal_volume_data)) = lo_eta_coeffs;
+    krivodonova(&nodal_scalar_data_to_limit, element, mesh, neighbor_data);
+    to_nodal_coefficients(&expected, expected_coeffs, mesh);
+    CHECK_ITERABLE_APPROX(get(nodal_scalar_data_to_limit), expected);
+  };
+
+  // Map between 2D and 1D coefficients:
+  // [(0,0), (1,0), (2,0), (0,1), (1,1), (2,1), (0,2), (1,2), (2,2)]
+  //  (0,     1,     2,     3,     4,     5,     6,     7,     8)
+
+  // Apply to (2,2)
```

Comment:
Could you document a little more verbosely what is happening here? Is each of these `test_X_Y` functions testing all the different ways in which the `(X,Y)` coeff can be modified? If so perhaps add a statement saying roughly that. Consider renaming to `test_limiting_of_X_Y_coeff`, or similar.